modules/core/src/test/java/org/apache/ignite/internal/processors/rest/RestProcessorInitializationTest.java

```diff
@@ -0,0 +1,224 @@
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testCustomRestProcessorInitialization() throws Exception {
+        IgniteConfiguration cfg = getConfiguration(getTestIgniteInstanceName(0));
+
+        cfg.setPluginProviders(new TestRestProcessorProvider());
+
+        IgniteEx ignite = startGrid(cfg);
+
+        assertEquals(ignite.context().rest().getClass(), TestGridRestProcessorImpl.class);
+    }
+
+    /**
+     * Test implementation of {@link PluginProvider} for obtaining {@link TestGridRestProcessorImpl}.
+     */
+    private static class TestRestProcessorProvider implements PluginProvider {
+        /** {@inheritDoc} */
+        @Override public String name() {
+            return "TEST_REST_PROCESSOR";
+        }
+
+        /** {@inheritDoc} */
+        @Override public String version() {
+            return null;
+        }
+
+        /** {@inheritDoc} */
+        @Override public String copyright() {
+            return null;
+        }
+
+        /** {@inheritDoc} */
+        @Override public void initExtensions(PluginContext ctx, ExtensionRegistry registry) {
+            // No-op.
+        }
+
+        /** {@inheritDoc} */
+        @Override public CachePluginProvider createCacheProvider(CachePluginContext ctx) {
+            return null;
+        }
+
+        /** {@inheritDoc} */
+        @Override public void start(PluginContext ctx) throws IgniteCheckedException {
+            // No-op.
+        }
+
+        /** {@inheritDoc} */
+        @Override public void stop(boolean cancel) throws IgniteCheckedException {
+            // No-op.
+        }
+
+        /** {@inheritDoc} */
+        @Override public void onIgniteStart() {
+            // No-op.
+        }
+
+        /** {@inheritDoc} */
+        @Override public void onIgniteStop(boolean cancel) {
+            // No-op.
+        }
+
+        /** {@inheritDoc} */
+        @Override public @Nullable Serializable provideDiscoveryData(UUID nodeId) {
+            return null;
+        }
+
+        /** {@inheritDoc} */
+        @Override public void receiveDiscoveryData(UUID nodeId, Serializable data) {
+            // No-op.
+        }
+
+        /** {@inheritDoc} */
+        @Override public void validateNewNode(ClusterNode node) throws PluginValidationException {
+            // No-op.
+        }
+
+        /** {@inheritDoc} */
+        @Nullable @Override public Object createComponent(PluginContext ctx, Class cls) {
+            if (cls.equals(IgniteRestProcessor.class))
+                return new TestGridRestProcessorImpl();
+
+            return null;
+        }
+
+        /** {@inheritDoc} */
+        @Override public IgnitePlugin plugin() {
+            return new IgnitePlugin() {
+                // No-op.
+            };
+        }
+    }
+
+    /**
+     * Test no-op implementation of {@link IgniteRestProcessor}.
+     */
+    private static class TestGridRestProcessorImpl implements IgniteRestProcessor {
```

Comment:
should use GridProcessorAdapter to avoid a boilerplate code

Reply:
Thanks! Done.